Source/PINDiskCache.m

```diff
@@ -881,8 +939,9 @@ - (id)objectForKeyedSubscript:(NSString *)key
             [self unlock];
             [self lockAndWaitForKnownState];
         }
-    
-        if (!self->_ttlCache || self->_ageLimit <= 0 || fabs([_metadata[key].date timeIntervalSinceDate:now]) < self->_ageLimit) {
+
+        NSTimeInterval ageLimit = _metadata[key].hasAgeLimit ? _metadata[key].ageLimit : self->_ageLimit;
+        if (!self->_ttlCache || ageLimit <= 0 || fabs([_metadata[key].date timeIntervalSinceDate:now]) < ageLimit) {
```

Comment:
Is this different behavior than above? I.e. it behaves like a ttlCache if there's an age limit *or* it's marked as a ttlCache here but not above?

Reply:
I think this is basically saying:
- If ttlCache == NO, return the object.
- If ttlCache == YES and ageLimit is not set, return the file.
- If ttlCache == YES, ageLimit is set and object is not expired return the object.